_posts/pl/2021-02-01-wszystko-co-chcielibyscie-wiedziec-o-fontach-ale-baliscie-sie-zapytac.md

```diff
@@ -0,0 +1,104 @@
+
+## Dołączanie fontów do naszej aplikacji
+
+Na koniec przyjrzyjmy się jest sposobom dołączania naszych fontów do projektu. Są różne możliwości, a kilka z nich, 
+razem z ich zaletami i wadami, przedstawiam poniżej:
+
+**@Font-face**
+
+Najbardziej znany i lubiany sposób definiowania nowych fontów, które chcemy użyć w naszym projekcie. Tworzymy nowy plik i wrzucamy do niego ścieżki do naszych fontów (najczęściej w formacie WOFF i WOFF2). Plik ma strukturę w której definiujemy nazwę i ścieżki do naszych plików:
+```css
+@font-face {
+  font-family: 'NowyFont';
+  src: url('nowyFont.woff2') format('woff2'),
+       url('nowyFont.woff') format('woff'),
+}
+```
+Struktura jest czytelna, więc możemy łatwo ją modyfikować i dodawać kolejne fonty. Aplikacja nie czeka aż fonty zostaną załadowane, więc nie musimy się martwić, że wydłużamy czas ładowania aplikacji. Może to być poręczne, ale powoduje również pewne problemy, ponieważ przez to mamy często do czynienia z 'mrugnięciem fontu', czyli z szybką zmianą wyglądu tekstu. Są różne strategie, radzenia sobie z tym przez przeglądarki. Najczęściej tekst jest na chwilę niewidzialny i jeżeli po pewnym czasie font nie zostanie załadowany, tekst się wyświetla, ale z użytym domyślnym fontem. W momencie kiedy font się załaduje, tekst 'wskakuje' we właściwy font. Czym font więcej waży i czym mamy gorszy transfer może to potrwać dłużej. Wszystkie żądania odbywają się niezależnie, więc ciężko je razem zgrupować, aby mogły załadować się w tym samym momencie.
+
+**font-display**
+
+Na ratunek z pojawiającym się nagle tekstem przychodzi własność font-display: swap, która możemy ustawić w naszym pliku @font-face. Dzięki niemu nie uświadczymy już dwóch mrugnięć (pojawienie się tekstu i zmiana fontu), a jednego: tekst pojawia się od razu i czeka, aż font zostanie załadowany. Zaletą tego rozwiania jest prostota w konfiguracji. Minus jest to, że tak naprawdę nie wiele nam to jeszcze daje.
+
+**Wczytywanie fontu przed renderowaniem zawartości strony**
+
+Innym sposobem radzenia sobie z ładowanie fontów jest dołączanie linku do naszego html bezpośrednio do fontu z którego chcemy skorzystać:
+```html
+    <link rel="preload" href="https://fonts.gstatic.com/s/opensans/v13/k3k702ZOKiLJc3WVjuplzBampu5_7CjHW5spxoeN3Vs.woff2" as="font" type="font/woff2" crossorigin>
+```
+Rozwiązanie jest łatwe w implementacji, jeden link i mamy wszystko czego potrzebujemy. Nie musimy się martwić, że teksty na nasze stronie będą źle wyglądać bez fontów. Rozwiązanie to może wydawać się bardzo dobre, za wyjątkiem jednej rzeczy. Przez to, że wczytujemy font wcześniej, opóźniamy wejście na stronę. Czas wejście na stronę jest dłuższy, ponieważ załadowanie fontu ma wyższy priorytet i odbędzie się przed wyświetleniem aplikacji.
```

Comment:
literówka
naszej stronie

proponuję też synonim do rzecz. Np. element, detal, kwestia. Może też warto połączyć te zdania, by razem zyskały trochę na tempie? Ale to zostawiam Twojemu wyczuciu.

Rozwiązanie to może wydawać się bardzo dobre, za wyjątkiem jednej kwestii, a mianowicie przez wczytywania fontu wcześniej, opóźniamy wejście na stronę.

+ literówka
Czas wejścia